modules/lb-vshn-hiera/main.tf

```
@@ -0,0 +1,22 @@
+locals {
+  instance_fqdns = formatlist("%s.${var.node_name_suffix}", var.lb_names)
+
+  lb_count = length(var.lb_names)
+
+  content = templatefile(
+    "${path.module}/templates/hieradata.yaml.tmpl",
+    {
+      "cluster_id"   = var.cluster_id
+      "api_vip"      = cidrhost(var.api_vip_network, 0)
+      "router_vip"   = cidrhost(var.router_vip_network, 0)
+      "api_secret"   = var.lb_cloudscale_api_secret
+      "internal_vip" = cidrhost(var.privnet_cidr, 100)
+      "nat_vip"      = cidrhost(var.nat_vip_network, 0)
+      "nodes"        = local.instance_fqdns
+      "backends" = {
+        "api"    = formatlist("etcd-%d.${var.node_name_suffix}", range(3))
```

This is an OCP4-ism, as OCP4 requires that each master has a DNS record `etcd-N`, we need to decide if we want to keep this hardcoded here, or make the API backends configurable in the module. This is not urgent, but we should keep it in mind.

On second thought, should be easy enough to just use the same pattern as for the router backends, simply do

```suggestion
        "api"    = var.api_backends[*]
```

here, and set the variable accordingly when instantiating the LBs.